src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/StructBuilder.java

```diff
@@ -48,12 +48,14 @@
     private final GroupLayout structLayout;
     private final Type structType;
     private final Deque<String> prefixElementNames;
+    private final String safeParameterName;
 
     StructBuilder(JavaSourceBuilder enclosing, String className, GroupLayout structLayout, Type structType) {
         super(enclosing, Kind.CLASS, className);
         this.structLayout = structLayout;
         this.structType = structType;
         prefixElementNames = new ArrayDeque<>();
+        safeParameterName = isEnclosedBySameName("x")? "x$" : "x";
```

Comment:
I'm not sure by this. It seems very tailored to the issue you have found. I guess I'd prefer a function like:
```
String safeParameterName(String paramName)
```

Which can be called by clients. Note that when we'll add support for calling function pointers (https://git.openjdk.java.net/panama-foreign/pull/456) structs will also have "functions" (which allow to call a function pointer); these function have more than one parameter, each of which could probably clash with any of the enclosing classes. So, in short, I think all parameter names in nested classes should probably be guarded by similar code.